### PR TITLE
Llama Support

### PR DESCRIPTION
**Pull request by yoazmenda**
### Initial support for llama models as local llm

image

### How to use:
1. Add the llama model file somewhere into the the project directory (so it can be referenced by path later)
2. run locally: (change inside package.json "proxy": "http://backend:7860" into "http://localhost:5003". Then "make run_backend" followed by "cd src/frontend  & npm install & npm start
4. Alternatively run using docker "make dev" (takes forever to respond - Appreciate ideas on how to make docker setup to answer in reasonable time, or maybe comment out in docker scenario?)
5. In the Langflow UI, choose LlamaCpp as you llm and point to your model file

Tested on MacBook Pro M1 using [Vicuna](https://vicuna.lmsys.org/) for llama model weights

Download the model [here](https://huggingface.co/eachadea/ggml-vicuna-13b-4bit/resolve/main/ggml-vicuna-13b-4bit.bin)

### CODE REVIEW

**Review by ogabrielluiz — APPROVED** [review]
LGTM

File: src/backend/langflow/interface/custom_lists.py
Comment (ogabrielluiz) [inline src/backend/langflow/interface/custom_lists.py:1]: You don't need to add this as LlamaCpp is part of the dunder all in llms dunder init.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]

File: src/backend/langflow/interface/custom_lists.py
Comment (ogabrielluiz) [reply src/backend/langflow/interface/custom_lists.py:1]: By the way, we just pushed a PR that fixes some things in the Chain nodes, could you test your implementation with those changes, please? 

I'll try it out here but loading llama is proving to be a bit troublesome here.

Comment (yoazmenda) [reply src/backend/langflow/interface/custom_lists.py:1]: @ogabrielluiz 
Thanks :)
Removed the unnecessary custom object initialization. 
Also merged latest changes into the PR and it works 👍

File: dev.Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@ FROM python:3.10-slim
 WORKDIR /app
 
 # Install Poetry
-RUN apt-get update && apt-get install gcc curl -y
+RUN apt-get update && apt-get install gcc g++ curl -y
```
Comment (yoazmenda) [inline dev.Dockerfile:6]: g++ required when llama cpp python lib is a dependency

File: src/backend/langflow/interface/custom_lists.py
Comment (ogabrielluiz) [reply src/backend/langflow/interface/custom_lists.py:1]: Awesome. I haven't been able to test. Most likely I got the wrong file or my laptop can't handle it hahha

File: dev.Dockerfile (hunk 1 above)
Comment (ogabrielluiz) [reply dev.Dockerfile:6]: LGTM

**Comment by ogabrielluiz** [comment]
@lolxdmainkaisemaanlu your problem is the proxy. 
You might need to run `make dev build=1` to get the new proxy.

**Comment by kaleavess** [comment]
Hi @yoazmenda I am trying to load the llama.cpp model in the langflow(not in docker) however it didnt recognize the path. Where is yours locate?

**Comment by yoazmenda** [comment]
Hi @kaleavess ,
I simply created a directory called "models" in the project root and point there "./models/model_name.bin".
If you are using Docker, don't make sure to mount this model file as well.

If you still can't manage, can you share more details about your setup and/or more errors?

**Comment by kaleavess** [comment]
Appreciate for @yoazmenda reply. 
Here is the details, I installed langflow 0.0.7.6 at default location in my VM with ubuntu 22.10. (langchain is 0.0.178)
The root location :
1. Langflow.py --> /home/ubuntu/.local/bin/langflow.py
2. Langflow site-packages --> /home/ubuntu/.local/lib/python3.10/site-packages/langflow

I placed the LLM(7B LLaMA.cpp/ GPT4ALL) in the below path:
1. /home/
2. /home/ubuntu/
3. /home/ubuntu/.local/
4. /home/ubuntu/.local/bin/
5. /home/ubuntu/.local/lib/python3.10/site-packages/langflow

Then run it without docker
Result--> Error building node LlamaCpp: 1 validation error for LlamaCpp__root__

BTW, in the previous version of langflow(0.0.7.4) got the same location error. However I deleted the self.ctx things then  it load the LLM correctly but got another error so it stop me go in deep.I hope there is enough information.

**Pull request merged**